# HG changeset patch
# User Rik <rik@octave.org>
# Date 1542822372 28800
#      Wed Nov 21 09:46:12 2018 -0800
# Node ID 0cebf81fcdc5cf453c1bb5d9f3f2dfc28bbab61d
# Parent  0e4b9df85361de2aecb465a3b9d8107e4628fb9b
Stop segfault when load-path directory does not exist (bug #55046).

* load-path.cc (load_path::update): Supply missing C string to invocation of
warning() function call which uses '%s' in format string.

diff --git a/libinterp/corefcn/load-path.cc b/libinterp/corefcn/load-path.cc
--- a/libinterp/corefcn/load-path.cc
+++ b/libinterp/corefcn/load-path.cc
@@ -361,17 +361,18 @@ namespace octave
 
     package_map.clear ();
 
     for (auto& di : dir_info_list)
       {
         bool ok = di.update ();
 
         if (! ok)
-          warning ("load-path: update failed for '%s', removing from path");
+          warning ("load-path: update failed for '%s', removing from path",
+                   di.dir_name.c_str ());
         else
           add (di, true, "", true);
       }
   }
 
   bool
   load_path::contains_canonical (const std::string& dir) const
   {
